dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -53,29 +54,90 @@ public sealed class AzureOpenAITextToImageService : ITextToImageService
         HttpClient? httpClient = null,
         ILoggerFactory? loggerFactory = null,
         string? apiVersion = null)
+        : this(deploymentName, modelId, loggerFactory)
     {
         Verify.NotNullOrWhiteSpace(apiKey);
-        Verify.NotNullOrWhiteSpace(deploymentName);
-
-        this._deploymentName = deploymentName;
 
-        if (modelId is not null)
+        var connectorEndpoint = !string.IsNullOrWhiteSpace(endpoint) ? endpoint! : httpClient?.BaseAddress?.AbsoluteUri;
+        if (connectorEndpoint is null)
         {
-            this.AddAttribute(AIServiceExtensions.ModelIdKey, modelId);
+            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
         }
-        this.AddAttribute(DeploymentNameKey, deploymentName);
 
-        this._logger = loggerFactory?.CreateLogger(typeof(AzureOpenAITextToImageService)) ?? NullLogger.Instance;
+        this._client = new(new Uri(endpoint), new AzureKeyCredential(apiKey), GetClientOptions(httpClient, apiVersion));
+    }
+
+    /// <summary>
+    /// Create a new instance of Azure OpenAI image generation service
+    /// </summary>
+    /// <param name="deploymentName">Deployment name identifier</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL</param>
+    /// <param name="credential">Token credentials, e.g. DefaultAzureCredential, ManagedIdentityCredential, EnvironmentCredential, etc.</param>
+    /// <param name="modelId">Model identifier</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="loggerFactory">The ILoggerFactory used to create a logger for logging. If null, no logging will be performed.</param>
+    /// <param name="apiVersion">Azure OpenAI Endpoint ApiVersion</param>
+    public AzureOpenAITextToImageService(
+        string deploymentName,
+        string endpoint,
+        TokenCredential credential,
+        string? modelId,
+        HttpClient? httpClient = null,
+        ILoggerFactory? loggerFactory = null,
+        string? apiVersion = null)
+        : this(deploymentName, modelId, loggerFactory)
```

Comment:
I don't think we need this constructor, the one that takes `OpenAIClient` should suffuce

Reply:
Ok, removed.